app/server/utils/metrics.py

```diff
@@ -0,0 +1,385 @@
+            .filter(*standard_payment_filters) \
+            .filter(*date_filter) \
+                .all()
+                
+        try:
+            last_day_transaction_average = daily_transaction_average[0].average
+            transaction_average_list = [
+                {'date': item.date.isoformat(), 'average': item.average} for item in daily_transaction_average
+            ]
+        except IndexError:
+            last_day_transaction_average = 0
+            transaction_average_list = [{'date': datetime.datetime.utcnow().isoformat(), 'average': 0} ]
+
+        return transaction_average_list
+
+    try:
+        master_wallet_balance = cached_funds_available()
+    except:
+        master_wallet_balance = 0
+
+    daily_transaction_volume, last_day, last_day_volume = calculate_daily_transaction_volume()
+    daily_disbursement_volume = calculate_daily_disbursement_volume()
+    daily_transaction_average = calculate_daily_transaction_average()
+    transfer_use_breakdown = calculate_transfer_use_breakdown()
+    total_distributed, total_spent, total_exchanged = calculate_transaction_totals()
+    total_beneficiaries, _, total_users, has_transferred_count, exhausted_balance_count = calculate_user_totals()
+
+
+    data = {
+        'total_distributed': total_distributed,
+        'total_spent': total_spent,
+        'total_exchanged': total_exchanged,
+        'has_transferred_count': has_transferred_count,
+        'zero_balance_count': exhausted_balance_count,
+        'total_beneficiaries': total_beneficiaries,
+        'total_users': total_users,
+        'master_wallet_balance': master_wallet_balance,
+        'daily_transaction_volume': daily_transaction_volume,
+        'daily_disbursement_volume': daily_disbursement_volume,
+        'daily_transaction_average': daily_transaction_average,
+        'transfer_use_breakdown': transfer_use_breakdown,
+        'last_day_volume': {'date': last_day.isoformat(), 'volume': last_day_volume},
+        'filter_active': filter_active
+    }
+
+    return data
+
+
+def apply_user_filters(query, filters):
+    if(filters is None):
+        return query
+
+    # if attribute is not CA table, do the required join and apply filters
+    for table_name, filts in filters.items():
+        if table_name == TransferAccount.__tablename__:
+            query = query.join(TransferAccount, TransferAccount.id == CreditTransfer.sender_transfer_account_id)
+            query = apply_single_column_filter(query, filts, TransferAccount)
+        elif table_name == CustomAttributeUserStorage.__tablename__:
+            query = apply_ca_pivot_table(query, filts)
+        elif table_name == User.__tablename__:
+            query = query.join(User, User.id == CreditTransfer.sender_user_id)
+            query = apply_single_column_filter(query, filts, User)
+    return query
+
+def apply_ca_pivot_table(query, filters):
+
+    # get all custom attributes and create pivot table
```

Comment:
pivots the custom attributes table so that all unique attributes turn into columns and the filters can be applied as you would for other attributes. (related functions not completely DRY yet)